### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by AlvaroVega**
- [x] Readme
- [x] DockerFile
- [x] docker-compose.yml

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+FROM centos:6
+
+MAINTAINER IoT team
+
+RUN yum update -y && yum install -y wget
+RUN wget http://dl.fedoraproject.org/pub/epel/6/x86_64/epel-release-6-8.noarch.rpm
+RUN yum localinstall -y --nogpgcheck epel-release-6-8.noarch.rpm
+RUN yum install -y rpm-build git java-1.7.0-openjdk java-1.7.0-openjdk-devel
```
Comment (LeandroGuillen) [inline Dockerfile]: You may want to group several RUN commands into one. This helps speedup the build process and takes less space.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+RUN sed -i "s/port: 8080/port: 7070/g" conf/config.yml
+RUN sed -i "s/port: 8081/port: 7071/g" conf/config.yml
+RUN sed -i "s/localhost/mysql/g" conf/config.yml
+
+ADD target/keypass-0.4.3.jar /data/keypass.jar
+ADD conf/config.yml /data/config.yml
+
+#RUN java -jar /data/keypass.jar db migrate /data/config.yml
+#CMD java -jar /data/keypass.jar server /data/config.yml
```
Comment (LeandroGuillen) [inline Dockerfile]: Leftover?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+db:
+  image: mysql
+  expose:
+    - "3306"
+  ports:
+    - "3307:3306"
```
Comment (LeandroGuillen) [inline docker-compose.yml]: Why do you expose this port to the host? And if you do, why use a different port than the default 3306?

Comment (AlvaroVega) [inline docker-compose.yml]: Because probably you already are running another mysql in that port as I do

File: Dockerfile (hunk 2 above)
Comment (AlvaroVega) [inline Dockerfile]: Still I'm not sure if that should be better here or https://github.com/telefonicaid/fiware-keypass/pull/60/files#diff-4e5e90c6228fd48698d074241c2ba760R23

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# Install Maven
+RUN wget -c http://ftp.cixug.es/apache/maven/maven-3/3.2.5/binaries/apache-maven-3.2.5-bin.zip
+RUN unzip -oq apache-maven-3.2.5-bin.zip
+RUN cp -rf apache-maven-3.2.5 /opt/maven
+RUN ln -fs /opt/maven/bin/mvn /usr/bin/mvn
+
+## Install MySQL (client at least?)
+RUN yum -y install mysql
+
+RUN mkdir /github && mkdir /github/telefonicaid
```
Comment (LeandroGuillen) [inline Dockerfile]: mkdir -p /github/telefonicaid

File: docker-compose.yml (hunk 3 above)
Comment (LeandroGuillen) [inline docker-compose.yml]: But why expose it to host?

Comment (AlvaroVega) [inline docker-compose.yml]: Just to provide a way to debug it

**Comment by XavierVal** [comment]
A readme file to launch the docker file would be useful, we can reuse the CB readme: 
https://raw.githubusercontent.com/telefonicaid/fiware-orion/develop/docker/README.md

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+WORKDIR /github/telefonicaid
+RUN git clone https://github.com/telefonicaid/fiware-keypass
```
Comment (dmoranj) [inline Dockerfile]: This isn't needed. Use a COPY instruction to add the local Github code to /opt/keypass

**Comment by AlvaroVega** [comment]
Now the problem is that keypass does not found db (mysql) host.

**Pull request merged**